plugins.md

```diff
@@ -0,0 +1,225 @@
+* Returns `Transaction`
+
+Creates and returns a new `Transaction` object. A `Transaction` object
+represents a root-level trace span, and exposes functions for adding labels,
+ending the span, and serializing the current trace context. Note that the
+underlying root span is started when this function is called.
+
+This function consults the trace agent's tracing policy to determine whether a
+trace span should actually be started or not. If the tracing policy prohibits
+a new span to be created under the current circumstances, this function will
+return `null` instead.
+
+#### `api.getTransaction()`
+* Returns `Transaction`
+
+Returns a Transaction object that corresponds to a root span started earlier
+in the same context, or `null` if one doesn't exist.
+
+#### `api.runInRootSpan(options, fn)`
+* `options`: [`TraceOptions`](trace-span-options)
+* `fn`: `function(Transaction): any`
+* Returns `any`
+
+Runs the given function in a root span corresponding to an incoming request,
+passing it the result of calling `api.createTransaction(options)`. The provided
+function should accept a nullable `Transaction` object. If `null` is provided,
+the function should proceed as if nothing is being traced.
+
+#### `transaction.runInChildSpan(options, fn)`
+* `options`: [`TraceOptions`](trace-span-options)
+* `fn`: `function(ChildSpan): any`
+* Returns `any`
+
+Runs the given function in a child span corresponding to an incoming request.
+The provided function should accept a `ChildSpan` object, which represents a
+child span.
+
+#### `transaction.addLabel(key, value) | childSpan.addLabel(key, value)`
+* `key`: `string`
+* `value`: `any`
+
+Add a label to the span associated with the calling object.
+
+#### `transaction.endSpan() | childSpan.endSpan()`
+
+Ends the span associated with the calling object. This function should only be
+called once.
+
+#### `api.runInChildSpan(options, fn)`
+* `options`: [`TraceOptions`](trace-span-options)
+* `fn`: `function(ChildSpan): any`
+* Returns `any`
+
+A shortcut for calling `api.getTransaction()` followed by
+`transaction.runInChildSpan(options, fn)`. Note that while this function accepts
+the same arguments as `transaction.runInChildSpan`, the function being passed in
+should handle being passed `null` as a parameter. This will be the case if
+`api.getTransaction` returns `null`.
+
+#### Trace Span Options
+
+Some functions above accept a `TraceOptions` object, which has the following
+fields:
+
+* `name`: `string`
+  * Required
+  * The name that should be given to the newly created span.
+* `traceContext`: `string`
+  * Optional for root spans, ignored for child spans
+  * A serialized trace context. If the module being traced is a web framework,
+    the plugin that patches this module should attempt to extract this from
+    an incoming request header and set this field. Omitting this field will
+    never cause an error, but may cause trace spans that correspond to a single
```

Comment:
Your replacement is addressed towards application developers but this document should address plugin developers. So while I understand the need to stress that omitting the trace context could cause services to appear dis-associated, I think this needs to be worded in a different way. I've revised the original text slightly to include some of the changes.